---
title: "SigFun: a functional analysis tool for signature."
package: SigFun

output:
  BiocStyle::html_document:
    toc: true
    toc_depth: 3
    number_sections: true
date: "`r Sys.Date()`"
abstract: >
    Biomarker exploration has become central to advancing clinical diagnostics and therapeutic decision-making, with a notable shift from single-gene markers to multigene signatures that better capture the complexity of biological processes. For example, in breast cancer, multigene signatures such as MammaPrint and Oncotype DX have been robustly validated and are now widely used to improve risk stratification and personalize adjuvant chemotherapy decisions. However, functional interpretation of these signatures remains challenging, especially when gene sets are small and traditional enrichment analyses are not feasible. To address this, we developed SigFun, a novel analytical framework that leverages transcriptome data to map multigene signatures ontothe whole genome, irrespective of whether signatures are binary or continuous. SigFun overcomes this by mapping signature genes to the entire transcriptome using correlation-based scoring or logistic regression. This creates a genome-wide association profile that preserves biological context while amplifying signal detection capability. By transforming signatures into transcriptome-wide quantitative traits, SigFun unlocks functional enrichment analysis even for signatures with as few as 5-10 genes â€” a capability we term scale-bridging analysis. This innovation allows researchers to uncover pathway-level insights from clinically validated but mechanistically opaque signatures, bridging the gap between biomarker utility and biological understanding.

vignette: >
  %\VignetteIndexEntry{SigFun}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo=FALSE, results='asis'} 
BiocStyle::markdown()
```

# Introduction {#intro}
## Biomarker exploration {#biomark}
Biomarker exploration in clinical practice represents a pivotal area ofclinical research, as it holds significant potential for enhancing diagnostic and therapeutic strategies. The trend in biomarker exploration isshifting towards the use of multi-gene signatures, which generally outperform single-gene markers due to their ability to provide a more comprehensive view of complex biological processes. This approach has been successfully applied in clinical settings, where several multi-gene signatures have been implemented to guide treatment decisions. Notable examples include MammaPrint, a 70-gene signature used in breast cancer prognosis, and OncoTypeDX, a 21-gene signature also employed in breast cancer management. These signatures have demonstrated the utility of multi-gene biomarkers in improving patient outcomes by offering more precise risk assessments and personalized treatment plans.

## Current limitation in deciphering multi-gene signatures {#limit}
Current limitations in the functional analysis of multi-gene signatures pose significant challenges in fully understanding their biological implications. While it is possible to individually interpret each gene within a signature, a systematic approach to deciphering these multi-gene signatures remains elusive. This issue is particularly pronounced for signatures consisting of a small set of genes, where conventional methods such as Over-Representation Analysis (ORA) and Gene Set Enrichment Analysis (GSEA) become infeasible. Consequently, this limitation hinders further advanced analysis of the signatures, particularly in elucidating their related biological functions or underlying mechanisms, thereby obstructing a deeper understanding of their roles in complex biological processes.
    
## SigFun: scale-bridge a signature to genome scale  {#scale-bridge}
SigFun is a novel solution designed to address the challenges associated with functional analysis of multi-gene signatures. SigFun leverages transcriptome data as a surrogate, allowing users to evaluate their signatures regardless of the size. Furthermore, the versatility of SigFun also allowing users to evaluate their signatures in various format types including binary values, such as the high or low risk classifications provided by MammaPrint, or continuous values, like prognostic risk scores. 

In the following tutorial, we will utilize a specific study and its corresponding dataset (GSE181574) as an illustrative example to demonstrate how SigFun can be employed to decipher the biologically relevant functions associated with the MammaPrint signature.

# Installation  {#install}
Here is the procedures of running the **SigFun** package on your system. We assume that you have already installed the R program (see the R project at [http://www.r-project.org](http://www.r-project.org)  and are familiar with it. You need to have R 4.5.0 or a later version installed for running SigFun.

Our package is available at the Bioconductor repository [http://www.bioconductor.org] (http://www.bioconductor.org). 
To install our package, first, you need to install the core Bioconductor packages. If you have already installed the Bioconductor packages on your system, you can skip the following step below.

```{r install_Bioconductor, eval=FALSE}
if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
BiocManager::install()
```

Once the core Bioconductor packages have been installed, we can begin the installation of the SigFun package.

```{r install_package, eval=FALSE}
if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
BiocManager::install("SigFun")
``` 

After conducting the above step, now you can load in our package and start using it!

```{r load, message=FALSE}
library(SigFun)
library(BiocStyle)
library(S4Vectors)
library(dplyr)
library(utils)
```

# Quick start with demo data {#quick}
We use GSE181574 to demonstrate how SigFun can help users to analyze biologicalfunctions related to a signature. GSE181574 is the gene expression profile of a breast cancer cohort which used a 70-gene signature to classify patients as high-risk or low-risk in tumor progression. Although the related biological function of every single gene can be annotated, respectively, a systematicallyfunctional assessment connecting the signature to tumor progression is not easy.As mentioned in the Introduction, SigFun takes the whole transcriptome expression as the surrogate to decipher related biological functions of the signature.

## Demo data description {#demo}
Let's load the demo data. There are two objects: (1) SE_GSE181574 (the expression profile and patient signature score) and (2) pathway.all (biological function and corresponding genes). 
The main input is a S4 class SummariedExperiment object:
```{r demo}
data("demo_GSE181574")
slotNames(SE_GSE181574)
```

The other input is a list which contains biological functions and their corresponding genes.
In this demo, the pathway.all was obtained from MsigDB which containing at least:
* GO-biological process (GOBP)
* GO-molecular function (GOMF)
* GO-cellular Component (GOCC)
* Kyoto Encyclopedia of Genes and Genomes (KEGG)
* MSigDB HALLMARK (HALLMARK)
* WikiPathways (WP)
* REACTOME Pathways (REACTOME)
* Signaling related Pathways (SIGNALING)

Although we provide a basic ontology database, users can replace it with their own gene set to align with their own interest. 
To see the list structure:
```{r pathway}
head(pathways.all)
```

## SigFun one-click command {#one-click}
To analyze the signature and obtain related biological function, we can just use sig2Fun (signature-to-functions) by setting corresponding parameters. 
In this demonstration, the MammaPrint signature classify patients into high- and low-risk (a binary variable). Consequently, the proper statistical method is a logistic regression model (cor.method="logit") and the signature does not need to be normalized (Z.transform=FALSE).

```{r SigFun, message=FALSE}
# Run signature-to-function analysis
res <- sig2Fun(
    SE_data=SE_GSE181574,
    ranking.method="stat",
    species="human",
    cor.method="logit",
    pathways_all=pathways.all,
    topN=10,
    Z.transform=FALSE,
    significat_type="pval",
    strings=c("HALLMARK")
)
```

## Visualization of results {#visual}
Users can visualize top signature-related function. 
The plot object(s) dependent on the parameter "strings" assigned in the sig2Fun function.
For example, when we assign strings=c("HALLMARK") and topN=10. sig2Fun will plot top 10 functions with strings containg "HALLMARK". On the other hand, if you assign topN=5 and strings="GOBP", sig2Fun will plot top 5 functions with strings containg "BOBP". Currently, we provide two ways of visualization: (1) barplot and (2) heatmap.

### Visualization 1: barplot
The barplot shows only top function names associated with the signature.
Functions are ordered by their significancy and colored by positive association in red and negative association in blue.
```{r barplot}
res$barplots[["HALLMARK"]]
```

### Visualization 2: functional enrichment heatmap
There are four columns in the heatmap (from left to right):
1. NES
2. p-value
2. Score distribution
3. Name of function
```{r heatmap}
res$heatmap[["HALLMARK"]]
```


#  Analyze your own dataset step-by-step in SigFun {#how}
Usually, users' raw datasets are in matrix or data.frame format. This tutorial will help users to construct their SummarizedExperiment objectusing their own datasets and run analysis step-by-step instead of one-click for more customized options.

## Input data: Constructing SummarizedExperiment object {#input}
First, we dissect the demo data to inspect their structures.
** assay: your expression profile (sample in column and gene in row)
** colData: signature score (the corresponding score assigned to each sample)
** rowData: gene annotation (ensemble ID and official gene symbol lookup table)
Note: In rowData, the "ensg_id" must align with the row name of GENE_MAT and "gene_symbol" must align with the gene ID system in pathway.all mentioned in the Demo data description.

```{r char_table}
# expression profile
GENE_MAT <- as.data.frame(SummarizedExperiment::assay(SE_GSE181574))
GENE_MAT[1:5,1:5]
# Signature score
SIG_MAT <- as.data.frame(SummarizedExperiment::colData(SE_GSE181574))
head(SIG_MAT)
# gene symbol mapping table 
mapping <- as.data.frame(SummarizedExperiment::rowData(SE_GSE181574))
head(mapping)
```

Now, we will construct SummarizedExperiment object from the three raw datasets and users can just replace them using their own dataset.
We will use the `SummarizedExperiment` function from `r Biocpkg("SummarizedExperiment")`. 
Follow the command below to create this object.
```{r build_se}
assay.data <- GENE_MAT
rowData.data <- mapping
colData.data <- SIG_MAT

SE_data <- SummarizedExperiment::SummarizedExperiment(
    assays=list(abundance=as.matrix(assay.data)),
    rowData=S4Vectors::DataFrame(rowData.data, 
    row.names=rowData.data$ensg_id), colData=colData.data)

SE_data
```


## sigCor: run gene correlation with signature patterns {#sigcor}
sigCor is the scale-bridging function which bridges the signature to whole genome scale. 
```{r SigCor}
# Method for calculate correlation
# logit for binary signature
# pearson or spearman for continuous data
cor.method="logit"
SE_data.cor <- sigCor(
    SE_data=SE_data, 
    cor.method=cor.method,
    Z.transform=FALSE)
```

## sig2GSEA: Run fGSEA using gene ranking values load corRES {#gsea}
The whole genome score of the signature could be assessed by sig2GSEA function.
```{r Sig2GSEA, message=FALSE}
# Ranking method for GSEA. (stat for statistics or pval for p-value)
ranking.method="stat"
SE_data.fgsea <- sig2GSEA(
    SE_data.cor=SE_data.cor,
    ranking.method=ranking.method,
    pathways.all=pathways.all)
```

## Visualization
### barplot
Here, users can visualize results using barplot.
Let's set "strings" to investigate signaling-specific pathways.
```{r plot.bar}
strings=c("REACTOME")
barplots <- plot_bar(
    SE_data.fgsea=SE_data.fgsea, 
    topN=10, 
    significat_type="pval", 
    strings=strings)
barplots
```

### heatmap
#### REACTOME Pathways
```{r plot_heat:REACTOME}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="REACTOME", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

#### Kyoto Encyclopedia of Genes and Genomes
```{r plot_heat:KEGG}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="KEGG", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

#### WikiPathways
```{r plot_heat:WP}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="WP", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

#### Gene Ontology: biological process
```{r plot_heat:GOBP}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="GOBP", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

#### Gene Ontology: molecular function
```{r plot_heat:GOMF}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="GOMF", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

#### Gene Ontology: cellular component
```{r plot_heat:GOCC}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="GOCC", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

#### Signal transduction associated function
```{r plot_heat:SIGNALING}
heatmap <- plot_heat(
    SE_data.fgsea=SE_data.fgsea, 
    strings="SIGNALING", 
    significat_type="pval", 
    topN=10, 
    pathways.all=pathways.all, 
    ranking.method=ranking.method)
heatmap
```

## Applications of SigFun {#applications}
Further applications of SigFun highlight its versatility in handling multiple types of signatures. We believe that SigFun's general applicability enables it to analyze various signature types effectively, including immune-related signatures, drug-response signatures, and tumor-progression signatures. This broad applicability positions SigFun as a powerful resource for researchers seeking to decipher the biological functions and clinical implications of diverse gene signatures across different disease contexts.

# Session info {#info}
```{r sessionInfo, echo=FALSE}
sessionInfo()
```
