<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualization Functions • SigFun</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Visualization Functions">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SigFun</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.11</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/QuickStart.html">QuickStart with Streamlined Workflow</a></li>
    <li><a class="dropdown-item" href="../articles/Data_Preparation.html">Data Preparation</a></li>
    <li><a class="dropdown-item" href="../articles/Stepwise_Workflow.html">Stepwise Workflow</a></li>
    <li><a class="dropdown-item" href="../articles/Visualization_Functions.html">Visualization Functions</a></li>
    <li><a class="dropdown-item" href="../articles/Custom_Gene_Ranking.html">Custom Gene Ranking</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/TMSWCChenglab/SigFun/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualization Functions</h1>
            
            <h4 data-toc-skip class="date">2025-11-10</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/TMSWCChenglab/SigFun/blob/HEAD/vignettes/Visualization_Functions.Rmd" class="external-link"><code>vignettes/Visualization_Functions.Rmd</code></a></small>
      <div class="d-none name"><code>Visualization_Functions.Rmd</code></div>
    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This guide helps you <strong>choose and customize</strong> figures
for interpreting GSEA results produced within
<strong>SigFun</strong>.<br>
For full workflows, see:</p>
<ul>
<li>
<strong>QuickStart</strong> — built-in analysis
(<code><a href="../reference/sig2Fun.html">sig2Fun()</a></code>)</li>
<li>
<strong>Stepwise Analysis</strong> — custom analysis pipeline</li>
<li>
<strong>Data Preparation</strong> — detailed input formatting</li>
<li>
<strong>Custom Gene Ranking</strong> — use your own ranking metric
before enrichment</li>
</ul>
</div>
<div class="section level2">
<h2 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h2>
<p>Before plotting, ensure you have:</p>
<ul>
<li>A <strong><code>SummarizedExperiment</code></strong> object with
compatible data and metadata<br>
</li>
<li>The <strong>GSEA result</strong> and <strong>correlation
table</strong> already added (automatically if you used
<code><a href="../reference/sig2Fun.html">sig2Fun()</a></code>)</li>
</ul>
<p>Below shows the <strong>minimal code</strong> to prepare the final
<code>SE_res</code> object for plotting using the built-in pipeline.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TMSWCChenglab/SigFun" class="external-link">SigFun</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load demo data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"expr.data"</span>, <span class="st">"mapping"</span>, <span class="st">"SIG_MAT"</span>, <span class="st">"t2g"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subset to KEGG and WikiPathways only (speeds up demo)</span></span>
<span><span class="va">t2g_kegg_wp</span> <span class="op">&lt;-</span> <span class="va">t2g</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^KEGG|^WP"</span>, <span class="va">gs_name</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">ensembl_gene</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Construct SummarizedExperiment</span></span>
<span><span class="va">SE_data</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span></span>
<span>  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">expr.data</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  rowData <span class="op">=</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="va">mapping</span>, row.names <span class="op">=</span> <span class="va">mapping</span><span class="op">$</span><span class="va">ensg_id</span><span class="op">)</span>,</span>
<span>  colData <span class="op">=</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="va">SIG_MAT</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run built-in SigFun analysis (see QuickStart for details)</span></span>
<span><span class="va">SE_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sig2Fun.html">sig2Fun</a></span><span class="op">(</span></span>
<span>  <span class="va">SE_data</span>,</span>
<span>  corMethod <span class="op">=</span> <span class="st">"logit"</span>,</span>
<span>  t2g <span class="op">=</span> <span class="va">t2g_kegg_wp</span>,</span>
<span>  strings <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEGG"</span>, <span class="st">"WP"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">#&gt; minSize, : For some of the pathways the P-values were likely overestimated. For</span></span>
<span><span class="co">#&gt; such pathways log2err is set to NA.</span></span>
<span><span class="co">#&gt; Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">#&gt; minSize, : For some pathways, in reality P-values are less than 1e-10. You can</span></span>
<span><span class="co">#&gt; set the `eps` argument to zero for better estimation.</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization-functions-12">Visualization Functions (12)<a class="anchor" aria-label="anchor" href="#visualization-functions-12"></a>
</h2>
<p>All visualization functions in this vignette take a
<code>SummarizedExperiment</code> object containing GSEA results. Each
function provides a different perspective on enriched pathways, from
overview summaries to detailed gene-level visualizations.</p>
<ol style="list-style-type: decimal">
<li><p><code><a href="../reference/plot_heat.html">plot_heat()</a></code> — Integrative multi-panel heatmap
summarizing pathways by NES, p-values, and enrichment curves.</p></li>
<li><p><code><a href="../reference/barPlot.html">barPlot()</a></code> — Bar chart ranking top enriched pathways
by significance or enrichment score.</p></li>
<li><p><code><a href="../reference/chordPlot.html">chordPlot()</a></code> — Circular diagram linking genes to
enriched pathways to show shared and unique memberships.</p></li>
<li><p><code><a href="../reference/cnetPlot.html">cnetPlot()</a></code> — Category–gene network visualizing
shared genes and connectivity among pathways.</p></li>
<li><p><code><a href="../reference/dotPlot.html">dotPlot()</a></code> — Dot chart encoding significance and gene
ratio across pathways.</p></li>
<li><p><code><a href="../reference/emapPlot.html">emapPlot()</a></code> — Enrichment map network showing
similarity between pathways based on shared genes.</p></li>
<li><p><code><a href="../reference/gseaPlot.html">gseaPlot()</a></code> — Per-pathway running enrichment score
curve with ranked gene metrics.</p></li>
<li><p><code><a href="../reference/heatPlot.html">heatPlot()</a></code> — Simple gene × pathway dot heatmap
encoding correlation and significance.</p></li>
<li><p><code><a href="../reference/lollipopPlot.html">lollipopPlot()</a></code> — NES-oriented lollipop chart
comparing enrichment magnitude and direction.</p></li>
<li><p><code><a href="../reference/ridgePlot.html">ridgePlot()</a></code> — Ridge density plot showing gene
ranking distributions within each pathway.</p></li>
<li><p><code><a href="../reference/treePlot.html">treePlot()</a></code> — Hierarchical clustering tree grouping
pathways by gene overlap or similarity.</p></li>
<li><p><code><a href="../reference/upsetPlot.html">upsetPlot()</a></code> — UpSet diagram illustrating
intersections of genes among enriched pathways.</p></li>
</ol>
<p>All visualization functions in this vignette take a
<strong>SummarizedExperiment</strong> object containing GSEA results.
Each function provides a different perspective on enriched pathways,
from overview summaries to detailed gene-level visualizations.</p>
<p><strong>Note:</strong> The following descriptions may not include all
parameters in each function. For details on available parameters and
customization options, refer to the function’s help page (e.g.,
<code><a href="../reference/barPlot.html">?barPlot</a></code> or <code><a href="../reference/barPlot.html">help("barPlot")</a></code>).</p>
<div class="section level3">
<h3 id="plot_heat">Integrative Heatmap — <code>plot_heat()</code><a class="anchor" aria-label="anchor" href="#plot_heat"></a>
</h3>
<p>The integrative heatmap illustrates the comprehensive overview of
enriched pathways across databases or keyword groups. Each panel
displays NES bars, enrichment curves, p-values, and pathway labels —
ideal for comparing multiple functional categories.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing pathway-level GSEA results.</td>
</tr>
<tr class="even">
<td>
<code>pathwaysAll</code> (required)</td>
<td>List of pathway-gene mappings used to define the panels.</td>
</tr>
<tr class="odd">
<td><code>strings</code></td>
<td>Keywords or database names (e.g., <code>"KEGG"</code>,
<code>"HALLMARK"</code>) to group panels.</td>
</tr>
<tr class="even">
<td><code>rankingMethod</code></td>
<td>Gene-level statistic used for ranking (<code>"cor"</code> or
other).</td>
</tr>
<tr class="odd">
<td><code>significantType</code></td>
<td>Statistical metric for filtering significance (<code>"pval"</code>
or <code>"padj"</code>).</td>
</tr>
<tr class="even">
<td><code>topN</code></td>
<td>Number of top enriched pathways to display per panel.</td>
</tr>
<tr class="odd">
<td><code>pvalueType</code></td>
<td>Statistical metric used to represent the values in significance
panel.</td>
</tr>
</tbody>
</table>
<p>Before generating the heatmap, prepare a pathway list mapping each
pathway to its member genes:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create named list of pathways with their genes</span></span>
<span><span class="va">pathways.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">t2g_kegg_wp</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">t2g_kegg_wp</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gs_name</span> <span class="op">==</span> <span class="va">x</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">ensembl_gene</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span>, </span>
<span>  nm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">t2g_kegg_wp</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Then, use <code><a href="../reference/plot_heat.html">plot_heat()</a></code> with
<code>pathways.all</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Panel: KEGG &amp; WikiPathways</span></span>
<span><span class="va">a_hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heat.html">plot_heat</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  pathwaysAll <span class="op">=</span> <span class="va">pathways.all</span>,</span>
<span>  strings <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEGG"</span>, <span class="st">"WP"</span><span class="op">)</span>,      <span class="co"># Users can also add other databases to plot accordingly</span></span>
<span>  rankingMethod <span class="op">=</span> <span class="st">"cor"</span>,   <span class="co"># Default is `cor` if streamlined workflow was ran</span></span>
<span>  significantType <span class="op">=</span> <span class="st">"pval"</span>,</span>
<span>  topN <span class="op">=</span> <span class="fl">10</span>, </span>
<span>  pvalueType <span class="op">=</span> <span class="st">"pvalue"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">a_hm</span><span class="op">$</span><span class="va">KEGG</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/kegg_and_wikipathways_heatmap-1.png" width="80%"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a_hm</span><span class="op">$</span><span class="va">WP</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/kegg_and_wikipathways_heatmap-2.png" width="80%"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Keyword focus (e.g., signaling)</span></span>
<span><span class="va">sig_hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heat.html">plot_heat</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>, </span>
<span>  pathwaysAll <span class="op">=</span> <span class="va">pathways.all</span>, </span>
<span>  strings <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SIGNALING"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">sig_hm</span><span class="op">$</span><span class="va">SIGNALING</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/signaling_heatmap-1.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="bar-plot-barplot">Bar Plot — <code>barPlot()</code><a class="anchor" aria-label="anchor" href="#bar-plot-barplot"></a>
</h3>
<p>The bar plot displays the top enriched pathways ranked by
<strong>NES</strong> by default, with bar opacity representing
<strong>−log10(p-value)</strong> by default to highlight the most
enriched categories. The plotting function also supports using
alternative variables for the x-axis and opacity.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing pathway-level GSEA results.</td>
</tr>
<tr class="even">
<td><code>x</code></td>
<td>Variable on the x-axis (<code>"NES"</code>, <code>"Count"</code>, or
<code>"GeneRatio"</code>).</td>
</tr>
<tr class="odd">
<td><code>alpha</code></td>
<td>Statistical significance metric used for opacity
(<code>"pvalue"</code>, <code>"p.adjust"</code>,
<code>"qvalue"</code>).</td>
</tr>
<tr class="even">
<td><code>topN</code></td>
<td>Number of pathways to plot.</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">res.bar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/barPlot.html">barPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"NES"</span>,</span>
<span>  alpha <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>  breaklineN <span class="op">=</span> <span class="fl">45</span></span>
<span><span class="op">)</span></span>
<span><span class="va">res.bar</span><span class="op">$</span><span class="va">barPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/bar_plot-1.png" width="80%"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.bar$tableBarPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="chord-diagram-chordplot">Chord Diagram — <code>chordPlot()</code><a class="anchor" aria-label="anchor" href="#chord-diagram-chordplot"></a>
</h3>
<p>The chord diagram Illustrates relationships between genes and their
enriched pathways from GSEA results. In this visualization, genes and
pathways are arranged as nodes around a circle, with connecting arcs
representing gene–pathway associations. This provides an intuitive
overview of shared and unique genes among pathways.</p>
<table class="table">
<colgroup>
<col width="37%">
<col width="62%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>GSEA results with gene-to-pathway mapping.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.chord</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chordPlot.html">chordPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  showCategory <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  breaklineN <span class="op">=</span> <span class="fl">30</span>, </span>
<span>  fontSize <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/chord_diagram-1.png" width="672"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.chord</span><span class="op">$</span><span class="va">chordPlot</span></span>
<span><span class="co"># table: res.chord$tableChordPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="concept-network-cnetplot">Concept Network — <code>cnetPlot()</code><a class="anchor" aria-label="anchor" href="#concept-network-cnetplot"></a>
</h3>
<p>The concept network visualizes gene–pathway relationships as a
network, showing overlap and connectivity between functional categories.
Nodes represent pathways and genes, while edges indicate membership
relationships. Optional color coding based on gene expression
correlation values highlights differential trends. This visualization
helps identify hub genes shared across multiple pathways and reveals the
interconnected structure of enriched gene sets.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing GSEA and optionally gene-level
correlation data.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
<tr class="odd">
<td><code>nodeLabel</code></td>
<td>Variable to label the node.</td>
</tr>
<tr class="even">
<td><code>color</code></td>
<td>Column name for coloring genes (e.g., <code>"cor"</code> or
<code>"log2FC"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.cnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cnetPlot.html">cnetPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  showCategory <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  nodeLabel <span class="op">=</span> <span class="st">'category'</span>,</span>
<span>  color <span class="op">=</span> <span class="st">"cor"</span>,</span>
<span>  fontSize <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>  breaklineN <span class="op">=</span> <span class="fl">30</span></span>
<span><span class="op">)</span></span>
<span><span class="va">res.cnet</span><span class="op">$</span><span class="va">cnetPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/concept_network-1.png" width="672"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># tables: res.cnet$tableEdgeCnetPlot, res.cnet$tableNodeCnetPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="dot-plot-dotplot">Dot Plot — <code>dotPlot()</code><a class="anchor" aria-label="anchor" href="#dot-plot-dotplot"></a>
</h3>
<p>The dot plot displays enriched pathways using dot size (gene
count/ratio) and color (significance or NES) for compact multi-encoding
visualization.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing GSEA results.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or specific pathways to show.</td>
</tr>
<tr class="odd">
<td><code>x</code></td>
<td>X-axis variable (<code>"GeneRatio"</code>, <code>"Count"</code>,
<code>"NES"</code>).</td>
</tr>
<tr class="even">
<td><code>size</code></td>
<td>Variable controlling dot size (<code>"Count"</code> or
<code>"GeneRatio"</code>).</td>
</tr>
<tr class="odd">
<td><code>alpha</code></td>
<td>Variable mapped to color/transparency (<code>"pvalue"</code>,
<code>"p.adjust"</code>, <code>"qvalue"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.dot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dotPlot.html">dotPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  showCategory <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"GeneRatio"</span>,</span>
<span>  size <span class="op">=</span> <span class="st">"Count"</span>,</span>
<span>  alpha <span class="op">=</span> <span class="st">'pvalue'</span>,</span>
<span>  breaklineN <span class="op">=</span> <span class="fl">20</span></span>
<span>  </span>
<span><span class="op">)</span></span>
<span><span class="va">res.dot</span><span class="op">$</span><span class="va">dotPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/dot_plot-1.png" width="672"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.dot$tableDotPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="enrichment-map-emapplot">Enrichment Map — <code>emapPlot()</code><a class="anchor" aria-label="anchor" href="#enrichment-map-emapplot"></a>
</h3>
<p>The enrichment map function generates an intuitive overview of the
functional landscape. It clusters related biological processes and
highlights pathway relationships based on shared genes. In this
network-based visualization, pathways are represented as nodes connected
by edges, where edge thickness reflects the degree of similarity between
pathways.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing pairwise pathway similarity.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
<tr class="odd">
<td><code>color</code></td>
<td>Node color variable (<code>"pvalue"</code>, <code>"p.adjust"</code>,
or <code>"qvalue"</code>).</td>
</tr>
<tr class="even">
<td><code>minEdge</code></td>
<td>Minimum pathway similarity required for drawing an edge in the
network.</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.ema</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emapPlot.html">emapPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>, </span>
<span>  showCategory <span class="op">=</span> <span class="fl">10</span>, </span>
<span>  color <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  minEdge <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="op">)</span></span>
<span><span class="va">res.ema</span><span class="op">$</span><span class="va">emapPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/enrichment_map-1.png" width="672"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.ema$tableEmapPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="enrichment-score-gseaplot">Enrichment Score — <code>gseaPlot()</code><a class="anchor" aria-label="anchor" href="#enrichment-score-gseaplot"></a>
</h3>
<p>The enrichmnet score function plots the running enrichment score,
ranked gene positions, and statistical metrics for selected gene sets.
It provides a detailed view of how individual genes contribute to the
overall enrichment signal within a pathway. This function wraps
<code><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html" class="external-link">enrichplot::gseaplot2</a></code>, streamlining the generation of
enrichment score visualizations.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment with GSEA running-score data.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
<tr class="odd">
<td><code>ESgeom</code></td>
<td>Geometry for enrichment curve (<code>"line"</code> or
<code>"dot"</code>).</td>
</tr>
<tr class="even">
<td><code>subPlots</code></td>
<td>Numeric integer between 1 to 3 to decide which subplots to display
(<code>1</code> = running score, <code>2</code> = gene ticks,
<code>3</code> = ranked list).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/gseaPlot.html">gseaPlot</a></span><span class="op">(</span>seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>, showCategory <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/enrichment_score-1.png" width="90%"></p>
</div>
<div class="section level3">
<h3 id="simple-dot-heatmap-heatplot">Simple Dot Heatmap — <code>heatPlot()</code><a class="anchor" aria-label="anchor" href="#simple-dot-heatmap-heatplot"></a>
</h3>
<p>The simple dot heatmap displays gene–pathway associations as a dot
matrix where color encodes correlation coefficient and size encodes
statistical significance measures.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment with gene-level statistics.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to display.</td>
</tr>
<tr class="odd">
<td><code>color</code></td>
<td>Column for dot color (<code>"cor"</code> or
<code>"log2FC"</code>).</td>
</tr>
<tr class="even">
<td><code>size</code></td>
<td>Column for dot size (<code>"pvalue"</code> or
<code>"p.adjust"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.heat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/heatPlot.html">heatPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  breaklineN <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  size <span class="op">=</span> <span class="st">'pvalue'</span>,</span>
<span>  color <span class="op">=</span> <span class="st">'cor'</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(Gene)`</span></span>
<span><span class="va">res.heat</span><span class="op">$</span><span class="va">heatPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/simple_dot_heatmap-1.png" width="672"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.heat$tableHeatPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="lollipop-plot-lollipopplot">Lollipop Plot — <code>lollipopPlot()</code><a class="anchor" aria-label="anchor" href="#lollipop-plot-lollipopplot"></a>
</h3>
<p>The lollipop plot displays normalized enrichment scores (or another
chosen metric) as horizontal lollipop bars, allowing quick comparison of
pathway direction, magnitude, and significance. The plotting function
also supports using alternative variables for the x-axis and
opacity.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing pathway-level statistics.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to show.</td>
</tr>
<tr class="odd">
<td><code>x</code></td>
<td>Variable mapped to the x-axis (<code>"NES"</code>,
<code>"Count"</code>, <code>"GeneRatio"</code>).</td>
</tr>
<tr class="even">
<td><code>alpha</code></td>
<td>Statistic controlling transparency (<code>"pvalue"</code>,
<code>"p.adjust"</code>, <code>"qvalue"</code>).</td>
</tr>
<tr class="odd">
<td><code>size</code></td>
<td>Variable for dot size (<code>"Count"</code> or
<code>"GeneRatio"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.lolli</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lollipopPlot.html">lollipopPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"NES"</span>,</span>
<span>  alpha <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  size <span class="op">=</span> <span class="st">"Count"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">res.lolli</span><span class="op">$</span><span class="va">lollipopPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/lollipop_plot-1.png" width="672"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.lolli$tableLollipopPlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ridge-plot-ridgeplot">Ridge Plot — <code>ridgePlot()</code><a class="anchor" aria-label="anchor" href="#ridge-plot-ridgeplot"></a>
</h3>
<p>The ridge plot shows the distribution of gene expression values (or
ranking metrics) for each enriched pathway. Each ridge represents a gene
set, with the curve shape reflecting the distribution of its member
genes along the ranked gene list. This visualization provides an
intuitive way to compare enrichment patterns across multiple
pathways.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment containing ranked gene statistics.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
<tr class="odd">
<td><code>fill</code></td>
<td>Variable mapped to ridge fill color (<code>"pvalue"</code>,
<code>"p.adjust"</code>, <code>"qvalue"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.ridge</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ridgePlot.html">ridgePlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>, </span>
<span>  showCategory <span class="op">=</span> <span class="fl">10</span>, </span>
<span>  fill <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  breaklineN <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span><span class="va">res.ridge</span><span class="op">$</span><span class="va">ridgePlot</span></span>
<span><span class="co">#&gt; Picking joint bandwidth of 0.219</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/ridge_plot-1.png" width="672"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.ridge$tableRidgePlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="tree-treeplot">Tree — <code>treePlot()</code><a class="anchor" aria-label="anchor" href="#tree-treeplot"></a>
</h3>
<p>The tree plot function generates a hierarchical clustering tree
(dendrogram) that groups enriched pathways based on gene overlap or
similarity. This visualization helps reveal functional relationships and
cluster structures among pathways. Each branch represents a group of
related pathways, while node colors can encode statistical significance
or other enrichment metrics. The function offers flexible control over
clustering methods, label formatting, highlighting, and layout
parameters for clear and customizable visualizations.</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment with pathway-level enrichment results.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
<tr class="odd">
<td><code>dotColor</code></td>
<td>Variable used to color pathway nodes (<code>"pvalue"</code>,
<code>"p.adjust"</code>, <code>"NES"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/treePlot.html">treePlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>, </span>
<span>  showCategory <span class="op">=</span> <span class="fl">10</span>, </span>
<span>  dotColor <span class="op">=</span> <span class="st">'pvalue'</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> # Invaild edge matrix for <span style="color: #0000BB;">&lt;phylo&gt;</span>. A <span style="color: #0000BB;">&lt;tbl_df&gt;</span> is returned.</span></span>
<span><span class="va">res.tree</span><span class="op">$</span><span class="va">treePlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/tree_plot-1.png" width="672"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.tree$tableTreePlot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="upset-plot-upsetplot">UpSet Plot — <code>upsetPlot()</code><a class="anchor" aria-label="anchor" href="#upset-plot-upsetplot"></a>
</h3>
<p>The upset plot visualizes gene overlap among multiple enriched
pathways using an UpSet layout. The lower intersection matrix indicates
which pathways contribute to each gene-set combination, while the upper
panel summarizes each intersection either as a <strong>boxplot</strong>
of correlation coefficients or a <strong>bar chart</strong> of set sizes
(controlled by <code>type = "box"</code> or <code>"bar"</code>).</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Essential Parameters</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<code>seDataFgsea</code> (required)</td>
<td>SummarizedExperiment with pathway-gene relationships.</td>
</tr>
<tr class="even">
<td><code>showCategory</code></td>
<td>Number or names of pathways to include.</td>
</tr>
<tr class="odd">
<td><code>type</code></td>
<td>Type of top summary plot (<code>"box"</code> or
<code>"bar"</code>).</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.upset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/upsetPlot.html">upsetPlot</a></span><span class="op">(</span></span>
<span>  seDataFgsea <span class="op">=</span> <span class="va">SE_res</span>, </span>
<span>  showCategory <span class="op">=</span> <span class="fl">5</span>, </span>
<span>  breaklineN <span class="op">=</span> <span class="fl">30</span>, </span>
<span>  type <span class="op">=</span> <span class="st">"box"</span><span class="op">)</span></span>
<span><span class="va">res.upset</span><span class="op">$</span><span class="va">upsetPlot</span></span></code></pre></div>
<p><img src="Visualization_Functions_files/figure-html/upset_plot-1.png" width="672"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table: res.upset$tableUpsetPlot</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualization-decision-guide">Visualization Decision Guide<a class="anchor" aria-label="anchor" href="#visualization-decision-guide"></a>
</h2>
<p>This section shows what each SigFun plotting function does, what it
needs, and how to use it quickly and consistently. Below is a
quick‑reference table summarizing all available visualization functions
in SigFun, their focus, and how to interpret them.</p>
<table class="table">
<colgroup>
<col width="25%">
<col width="25%">
<col width="25%">
<col width="25%">
</colgroup>
<thead><tr class="header">
<th>Visualization Function</th>
<th>Primary Focus</th>
<th>Best For</th>
<th>Typical Interpretation</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong><code><a href="../reference/plot_heat.html">plot_heat()</a></code></strong></td>
<td>Integrated multi-panel overview</td>
<td>Summarizing top pathways across multiple databases</td>
<td>Compare NES, p-values, and enrichment curves together.</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/barPlot.html">barPlot()</a></code></strong></td>
<td>Ranked significance summary</td>
<td>Highlighting top enriched pathways</td>
<td>Identify most significant or strongly enriched pathways.</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/chordPlot.html">chordPlot()</a></code></strong></td>
<td>Gene–pathway overlap</td>
<td>Showing shared or unique genes</td>
<td>Explore how genes contribute to multiple pathways.</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/cnetPlot.html">cnetPlot()</a></code></strong></td>
<td>Gene–pathway network</td>
<td>Visualizing connections among pathways</td>
<td>Reveal hub genes shared between categories.</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/dotPlot.html">dotPlot()</a></code></strong></td>
<td>Compact pathway summary</td>
<td>Quick overview of enrichment strength and gene ratios</td>
<td>Compare significance and gene ratio across terms.</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/emapPlot.html">emapPlot()</a></code></strong></td>
<td>Pathway similarity network</td>
<td>Detecting functional clusters</td>
<td>Identify groups of related pathways via shared genes.</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/gseaPlot.html">gseaPlot()</a></code></strong></td>
<td>Individual pathway enrichment score</td>
<td>Inspecting enrichment dynamics</td>
<td>Examine running enrichment score and cor e-enriched genes.</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/heatPlot.html">heatPlot()</a></code></strong></td>
<td>Gene × pathway association</td>
<td>Gene-level contribution visualization</td>
<td>Assess correlation or log2FC patterns across pathways.</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/lollipopPlot.html">lollipopPlot()</a></code></strong></td>
<td>NES-based summary</td>
<td>Comparing enrichment direction and magnitude</td>
<td>Visualize up- vs. down -regulated pathways by NES.</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/ridgePlot.html">ridgePlot()</a></code></strong></td>
<td>Distribution of ranked genes</td>
<td>Exploring enrichment density per pathway</td>
<td>View how ranked genes distribute across pathways.</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/treePlot.html">treePlot()</a></code></strong></td>
<td>Pathway clustering</td>
<td>Identifying similarity structure among terms</td>
<td>See hierarchical relationships among functional sets.</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/upsetPlot.html">upsetPlot()</a></code></strong></td>
<td>Gene set intersections</td>
<td>Comparing overlap among pathways</td>
<td>Quantify shared and unique genes between top terms.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: CentOS Stream 9</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Taipei</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_4.0.0    dplyr_1.1.4      SigFun_0.99.11   BiocStyle_2.34.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;   [3] rstudioapi_0.17.1           jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt;   [5] magrittr_2.0.3              ggtangle_0.0.7             </span></span>
<span><span class="co">#&gt;   [7] farver_2.1.2                rmarkdown_2.29             </span></span>
<span><span class="co">#&gt;   [9] GlobalOptions_0.1.2         fs_1.6.6                   </span></span>
<span><span class="co">#&gt;  [11] zlibbioc_1.52.0             ragg_1.3.3                 </span></span>
<span><span class="co">#&gt;  [13] vctrs_0.6.5                 memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [15] ggtree_3.14.0               rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [17] htmltools_0.5.8.1           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt;  [19] forcats_1.0.0               broom_1.0.10               </span></span>
<span><span class="co">#&gt;  [21] SparseArray_1.6.2           Formula_1.2-5              </span></span>
<span><span class="co">#&gt;  [23] gridGraphics_0.5-1          sass_0.4.10                </span></span>
<span><span class="co">#&gt;  [25] bslib_0.9.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [27] desc_1.4.3                  plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [29] cachem_1.1.0                igraph_2.1.4               </span></span>
<span><span class="co">#&gt;  [31] lifecycle_1.0.4             pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [33] Matrix_1.7-4                R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [35] fastmap_1.2.0               gson_0.1.0                 </span></span>
<span><span class="co">#&gt;  [37] GenomeInfoDbData_1.2.13     MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt;  [39] digest_0.6.37               aplot_0.2.8                </span></span>
<span><span class="co">#&gt;  [41] enrichplot_1.26.6           ggnewscale_0.5.2           </span></span>
<span><span class="co">#&gt;  [43] colorspace_2.1-1            patchwork_1.3.2            </span></span>
<span><span class="co">#&gt;  [45] AnnotationDbi_1.68.0        S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [47] textshaping_1.0.3           GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [49] RSQLite_2.4.3               org.Hs.eg.db_3.20.0        </span></span>
<span><span class="co">#&gt;  [51] ggpubr_0.6.1                labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [53] httr_1.4.7                  abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [55] compiler_4.4.3              bit64_4.6.0-1              </span></span>
<span><span class="co">#&gt;  [57] withr_3.0.2                 S7_0.2.0                   </span></span>
<span><span class="co">#&gt;  [59] backports_1.5.0             BiocParallel_1.40.2        </span></span>
<span><span class="co">#&gt;  [61] carData_3.0-5               DBI_1.2.3                  </span></span>
<span><span class="co">#&gt;  [63] R.utils_2.13.0              ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt;  [65] rappdirs_0.3.3              DelayedArray_0.32.0        </span></span>
<span><span class="co">#&gt;  [67] tools_4.4.3                 ape_5.8-1                  </span></span>
<span><span class="co">#&gt;  [69] R.oo_1.27.1                 glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [71] nlme_3.1-168                GOSemSim_2.32.0            </span></span>
<span><span class="co">#&gt;  [73] grid_4.4.3                  reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [75] fgsea_1.32.4                generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [77] gtable_0.3.6                R.methodsS3_1.8.2          </span></span>
<span><span class="co">#&gt;  [79] tidyr_1.3.1                 data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [81] car_3.1-3                   XVector_0.46.0             </span></span>
<span><span class="co">#&gt;  [83] BiocGenerics_0.52.0         ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt;  [85] pillar_1.11.0               stringr_1.5.2              </span></span>
<span><span class="co">#&gt;  [87] yulab.utils_0.2.1           circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [89] splines_4.4.3               treeio_1.30.0              </span></span>
<span><span class="co">#&gt;  [91] lattice_0.22-7              bit_4.6.0                  </span></span>
<span><span class="co">#&gt;  [93] tidyselect_1.2.1            GO.db_3.20.0               </span></span>
<span><span class="co">#&gt;  [95] Biostrings_2.74.1           knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [97] bookdown_0.44               IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [99] SummarizedExperiment_1.36.0 stats4_4.4.3               </span></span>
<span><span class="co">#&gt; [101] xfun_0.53                   Biobase_2.66.0             </span></span>
<span><span class="co">#&gt; [103] matrixStats_1.5.0           stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [105] UCSC.utils_1.2.0            lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [107] ggfun_0.2.0                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt; [109] evaluate_1.0.5              codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [111] tibble_3.3.0                qvalue_2.38.0              </span></span>
<span><span class="co">#&gt; [113] BiocManager_1.30.26         ggplotify_0.1.2            </span></span>
<span><span class="co">#&gt; [115] cli_3.6.5                   systemfonts_1.2.3          </span></span>
<span><span class="co">#&gt; [117] jquerylib_0.1.4             Rcpp_1.1.0                 </span></span>
<span><span class="co">#&gt; [119] GenomeInfoDb_1.42.3         png_0.1-8                  </span></span>
<span><span class="co">#&gt; [121] parallel_4.4.3              pkgdown_2.1.3              </span></span>
<span><span class="co">#&gt; [123] blob_1.2.4                  clusterProfiler_4.14.6     </span></span>
<span><span class="co">#&gt; [125] DOSE_4.0.1                  viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [127] tidytree_0.4.6              scales_1.4.0               </span></span>
<span><span class="co">#&gt; [129] ggridges_0.5.7              purrr_1.1.0                </span></span>
<span><span class="co">#&gt; [131] crayon_1.5.3                rlang_1.1.6                </span></span>
<span><span class="co">#&gt; [133] cowplot_1.2.0               fastmatch_1.1-6            </span></span>
<span><span class="co">#&gt; [135] KEGGREST_1.46.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Wei-Chung Cheng, Chia-Hsin Liu, Pei-Chun Shen, Wen-Jen Lin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
